symengine/printer.h

```diff
@@ -50,6 +50,29 @@ class Precedence : public BaseVisitor<Precedence> {
         }
     }
 
+    void bvisit(const UnivariatePolynomial &x) {
+        if (x.get_dict().size() == 1) {
+            auto it = x.get_dict().begin();
+            if (it->second == 0) {
+                precedence = PrecedenceEnum::Atom;
+            } else if (it->second == 1) {
+                if (it->first == 0 or it->first == 1) {
+                    precedence = PrecedenceEnum::Atom;
+                } else {
+                    precedence = PrecedenceEnum::Pow;
+                }
+            } else {
+                if (it->first == 0 and (it->second.get_basic()->__cmp__(*Expression(0).get_basic()) >= 0)) {
```

Comment:
If `it->first == 0` then this should be the precedence value of `it->second.get_basic()`.